lib/listener_manager.c

```diff
@@ -90,44 +93,137 @@ bool mptcpd_lm_listen(struct l_hashmap *lm,
                 l_error("Unable to bind MPTCP listener: %s",
                         strerror(errno));
                 (void) close(fd);
-                return false;
+                return -1;
         }
 
         if (listen(fd, 0) == -1) {
                 l_error("Unable to listen on MPTCP socket: %s",
                         strerror(errno));
                 (void) close(fd);
+                return -1;
+        }
+
+        return fd;
```

Comment:
One thing to consider: what if the caller leaves the port as `0` to request any open port. Would it then make sense to propagate the allocated port number back up to the caller, and make sure that port number is used in the hash?

Reply:
(This is in reference to the function as a whole - the chunk of the diff above isn't super relevant other than belonging to the `open_listener()` function)

Reply:
Commit f915775 provides the ephemeral port tracking you describe above.